### PR TITLE
Sorting

### PR DESCRIPTION
**Pull request by gusfel**
finished setting up the functionality of the 3 sort options for the review list.
newest and helpful are straightforward, but for relevant i created a formula that calculates the difference between today and the review date then divides that by the helpfulness score.  i then use that value to compare the reviews.

### CODE REVIEW

File: Reviews/src/Components/ReviewApp.jsx [hunk 1]
```diff
@@ -37,11 +41,69 @@ class ReviewApp extends React.Component {
...
+      if (!sortedRevs.length) {
+        sortedRevs.push(review);
+      } else {
+        let entered = false;
+        for (let j = 0; j < sortedRevs.length; j += 1) {
+          const sortedRev = sortedRevs[j];
+          if (sort === 'help') {
+            if (review.helpfulness > sortedRev.helpfulness && !entered) {
+              sortedRevs.splice(j, 0, review);
+              entered = true;
```
Comment (laenalea) [inline Reviews/src/Components/ReviewApp.jsx:76]: Might want to add a break here, bc if your current review is more helpful than the most helpful one, then you don't have to keep iterating over sortedRevs (also won't need the entered variable to help you from entering the if condition.

File: Reviews/src/Components/ReviewApp.jsx [hunk 2]
```diff
@@ -37,11 +41,69 @@ class ReviewApp extends React.Component {
...
+            }
+          } else if (sort === 'date') {
+            if (review.date < sortedRev.date && !entered) {
+              sortedRevs.splice(j, 0, review);
```
Comment (laenalea) [inline Reviews/src/Components/ReviewApp.jsx:80]: The logic would have to reverse to get the most recent (greater date). Should be review.date > sortedRev.date in order to sort most recent reviews at the beginning of your array

As of right now, your oldest reviews are showing at the beginning of your array.

File: Reviews/src/Components/ReviewApp.jsx [hunk 3]
```diff
@@ -37,11 +41,69 @@ class ReviewApp extends React.Component {
...
+            }
+          } else if (sort === 'relevant') {
+            const reviewDate = new Date(review.date);
+            const sortedRevDate = new Date(sortedRev.date);
+            const today = new Date();
+            const reviewDif = today - reviewDate;
+            const sortedDif = today - sortedRevDate;
+            const reviewRel = reviewDif / review.helpfulness
+            const sortedRel = sortedDif / sortedRev.helpfulness;
+            if (reviewRel > sortedRel && !entered) {
```
Comment (laenalea) [inline Reviews/src/Components/ReviewApp.jsx:91]: Following the logic here, the most relevant would be (smallest time difference)/(most helpful), for which the fraction would be approaching the limit of 0. In this case, that means the most relevant reviews would be the smallest, so your condition should be reviewRel < sortedRel in order to show your most relevant reviews at the beginning of your array.

**Review by laenalea — COMMENTED** [review]
Great job, Agustin! Just a few minor tweaks with the logic with your sorting algorithm, but fantastic work.

File: Reviews/src/Components/ReviewApp.jsx (hunk 2 above)
Comment (momo1113) [reply Reviews/src/Components/ReviewApp.jsx:80]: Agree with Anna, UI was seems right, 2020-08-05 appears on top of 2020-12-26

File: Reviews/src/Components/ReviewApp.jsx [hunk 4]
```diff
@@ -37,11 +41,69 @@ class ReviewApp extends React.Component {
...
+              entered = true;
+            }
+          } else if (sort === 'date') {
+            if (review.date < sortedRev.date && !entered) {
+              sortedRevs.splice(j, 0, review);
+              entered = true;
+            }
+          } else if (sort === 'relevant') {
+            const reviewDate = new Date(review.date);
+            const sortedRevDate = new Date(sortedRev.date);
```
Comment (momo1113) [inline Reviews/src/Components/ReviewApp.jsx:85]: SortedRev is an array, so here sortedRev array will get a new date?

**Review by jincheng411 — APPROVED** [review]
Seems like Anna made all the points

**Review by momo1113 — APPROVED** [review]

File: Reviews/src/Components/ReviewApp.jsx (hunk 1 above)
Comment (gusfel) [reply Reviews/src/Components/ReviewApp.jsx:76]: the break is taken care of by the entered = true and !entered

File: Reviews/src/Components/ReviewApp.jsx (hunk 2 above)
Comment (gusfel) [reply Reviews/src/Components/ReviewApp.jsx:80]: thanks for catching this!

File: Reviews/src/Components/ReviewApp.jsx (hunk 3 above)
Comment (gusfel) [reply Reviews/src/Components/ReviewApp.jsx:91]: good catch!

File: Reviews/src/Components/ReviewApp.jsx (hunk 4 above)
Comment (gusfel) [reply Reviews/src/Components/ReviewApp.jsx:85]: sortedRevs is the array while sortedRev is one of the reviews in the array

**Pull request closed without merging**